pulsar-common/src/test/java/org/apache/pulsar/common/util/keystoretls/JettySslContextFactoryWithAutoRefreshTest.java

```diff
@@ -0,0 +1,199 @@
+        int unusedPort = getUnusedPort();
+        connector.setPort(unusedPort);
+        connectors.add(connector);
+        server.setConnectors(connectors.toArray(new ServerConnector[0]));
+        server.start();
+        Thread.sleep(5000);
+        // client connect
+        HttpClientBuilder httpClientBuilder = HttpClients.custom();
+        RegistryBuilder<ConnectionSocketFactory> registryBuilder = RegistryBuilder.create();
+        registryBuilder.register("https", new SSLConnectionSocketFactory(getClientSslContext(),
+                new String[]{"TLSv1.2"}, new String[]{"TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384"}, new NoopHostnameVerifier()));
+        PoolingHttpClientConnectionManager cm = new PoolingHttpClientConnectionManager(registryBuilder.build());
+        httpClientBuilder.setConnectionManager(cm);
+        CloseableHttpClient httpClient = httpClientBuilder.build();
+        HttpGet httpGet = new HttpGet("https://localhost:" + unusedPort);
+        httpClient.execute(httpGet);
+    }
+
+    private static SSLContext getClientSslContext() {
+        return getSslContext(Resources.getResource("ssl/jetty_client_key.jks").getPath(),
+                "jetty_client_pwd",
+                Resources.getResource("ssl/jetty_client_trust.jks").getPath(),
+                "jetty_client_pwd");
+    }
+
+    private static SSLContext getSslContext(String keyStorePath, String keyStorePassword,
+                                            String trustStorePath, String trustStorePassword) {
+        try {
+            SSLContext sslContext = SSLContext.getInstance("TLS");
+            // key store
+            KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+            KeyStore keyStore = KeyStore.getInstance("JKS");
+            try (FileInputStream inputStream = new FileInputStream(keyStorePath)) {
+                keyStore.load(inputStream, keyStorePassword.toCharArray());
+            }
+            keyManagerFactory.init(keyStore, keyStorePassword.toCharArray());
+            KeyManager[] keyManagers = keyManagerFactory.getKeyManagers();
+            // trust store
+            TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+            KeyStore trustStore = KeyStore.getInstance("JKS");
+            try (FileInputStream inputStream = new FileInputStream(trustStorePath)) {
+                trustStore.load(inputStream, trustStorePassword.toCharArray());
+            }
+            trustManagerFactory.init(trustStore);
+            sslContext.init(keyManagers, trustManagerFactory.getTrustManagers(), new SecureRandom());
+            return sslContext;
+        } catch (Exception e) {
+            log.error("load ssl context error ", e);
+            return null;
+        }
+    }
+
+    private static int getUnusedPort() throws Exception {
```

Comment:
we already have a utility method for this

Reply:
@eolivelli Could you please tell me, where the method exists? I have searched for `unusedPort` and `freePort` etc. :)

Reply:
@eolivelli I fould `getFreePort` in `PulsarMultiListenersWithInternalListenerNameTest` and `PulsarMultiHostClientTest`.
But now we don't have module like `test-common` or `test-util`. So I suggest now we put this method here. And start a new PR for `test-util` WDYT?

Reply:
you can pass 0 and Jetty will get an ephemeral port, then you can ask to Jetty the actual LocalPort

Reply:
no need to add this method

Reply:
fixed